Add troubleshooting to windows install guide (#28211)
dotnet/docs: docs/core/install/windows.md

For more information about using .NET in a Docker container, see [Introduction to .NET and Docker](../docker/introduction.md) and [Samples](https://github.com/dotnet/dotnet-docker/blob/main/samples/README.md).

## Troubleshooting

> After installing .NET, you may run into problems trying to run .NET. This section collects those common problems and provides solutions.

Reviewer: ```suggestion
After installing .NET, you may run into problems trying to run .NET CLI commands. This section collects those common problems and provides solutions.
```

- [Can't find/run/locate .NET after installing](#cant-findrunlocate-net-after-installing)

> ### Can't find/run/locate .NET after installing

Reviewer: ```suggestion
### Can't find/run/locate the .NET SDK after installing
```

> 

Reviewer: should you show what error message they get? Here's an example. (it would be good to confirm the latest)
![image](https://user-images.githubusercontent.com/12971179/153519729-3ecd00ce-1508-46e5-9e1f-158c24a9293c.png)

Author: I wasn't sure if this was the exact error message. Looking through some of the linked issues I found a few different "affects" people had, some of them this error, others just having conflicts with no error, just knowing they're going to the wrong SDK or something.

Reviewer: I'd still add this one as an example. People will search for the error message and then would find this docs page if the text is part of the article not as an image.

> Most likely you've installed both the x86 (32-bit) and x64 (64-bit) versions of .NET. This is causing a conflict because when you run the `dotnet` command it's resolving to the x86 version when it should resolve to the x64 version. This is usually fixed by adjusting the `%PATH%` variable to resolve the x64 version first.

Reviewer: ```suggestion
Most likely you've installed both the x86 (32-bit) and x64 (64-bit) versions of the .NET SDK. This is causing a conflict because when you run the `dotnet` command it's resolving to the x86 version when it should resolve to the x64 version. This is usually fixed by adjusting the `%PATH%` variable to resolve the x64 version first.
```

> 

Reviewer: The above is correct for x64 systems. The opposite can happen for x86 as well (although more rare). But it would be good to write the instructions on a way that is more generic.

Author: @mairaw How could this happen on a x86 system? You can't install 64-bit apps on an 32-bit OS.

> 01. Verify that you have both versions installed by running the `where.exe dotnet` command, you should see the x86 version listed first:

Reviewer: ```suggestion
01. Verify that you have both versions installed by running the `where.exe dotnet` command. If you do, you should see your machine architecture listed first. This issue happens when the other version is listed first like in the following example running on a x64 machine:
```

Author: This isn't exactly true because it doesn't show a 64 folder to indicate the architecture like it does with 86. How about
```suggestion
01. Verify that you have both versions installed by running the `where.exe dotnet` command. If you do, you should see an entry for both the _Program Files\\_ and _Program Files (x86)\\_ folders. If the _Program Files (x86)\\_ folder is first as indicated by the following example, it's incorrect and you should continue on to the next step.
```

> 

Reviewer: I think sometimes even VS installs these things

    ```cmd
    > where.exe dotnet
    C:\Program Files (x86)\dotnet\dotnet.exe  
    C:\Program Files\dotnet\dotnet.exe
    ```

> 

Author: ```suggestion

    If it's correct and the _Program Files\\_ is first, you don't have the problem this section is discussing and you should create a [.NET help request issue on GitHub](https://github.com/dotnet/core/issues/new?assignees=&labels=&template=01_bug_report.md&title=)

```

01. Press the Windows button and type "Edit the system environment variables" into search. Select **Edit the system environment variables**.

    :::image type="content" source="media/windows/start-menu.png" alt-text="Windows start menu with edit environment variable":::

> 01. The **System Properties** window opens up to the **Advanced Tab**. Press **Enivornment Variables...**.

Reviewer: ```suggestion
01. The **System Properties** window opens up to the **Advanced Tab**. Select **Environment Variables**.
```

    :::image type="content" source="media/windows/system-props.png" alt-text="The Windows system properties panel open.":::

> 01. On the **Environment Variables** window, press the **Edit...** button under the **System variables** group.

Reviewer: ```suggestion
01. On the **Environment Variables** window, under the **System variables** group, select the *Path** row and then select the **Edit** button.
```

>     :::image type="content" source="media/windows/edit-vars.png" alt-text="The environment variables window with user and system variables.":::

Reviewer: wrong image?

Author: ```suggestion
    :::image type="content" source="media/windows/list-vars.png" alt-text="The environment variables window with user and system variables.":::
```

> 01. Use the **Move Up** and **Move Down** buttons to move the **C:\\Program Files\\dotnet\\** entry above **C:\\Program Files (x86)\\dotnet\\**.

Reviewer: markdown here looks funky?

Author: Preview is rendering OK. I think the github editor just has a hard time with escaped `\\` near `**`

For more information about using .NET in a Docker container, see [Introduction to .NET and Docker](../docker/introduction.md) and [Samples](https://github.com/dotnet/dotnet-docker/blob/main/samples/README.md).

## Troubleshooting

After installing the .NET SDK, you may run into problems trying to run .NET CLI commands. This section collects those common problems and provides solutions.

- [Can't find/run/locate .NET after installing](#cant-findrunlocate-the-net-sdk-after-installing)

> ### Can't find/run/locate the .NET SDK after installing

Author: ```suggestion
### It was not possible to find any installed .NET Core SDKs
```
